src/lib/operators/join_nested_loop.cpp

```diff
@@ -0,0 +1,254 @@
+          _right_matches.insert(RowID{chunk_id_right, right_value.chunk_offset()});
+        }
+      }
+    }
+  }
+}
+
+void JoinNestedLoop::_perform_join() {
+  auto left_table = _left_in_table;
+  auto right_table = _right_in_table;
+
+  auto left_column_id = _left_column_id;
+  auto right_column_id = _right_column_id;
+
+  if (_mode == JoinMode::Right) {
+    // for Right Outer we swap the tables so we have the outer on the "left"
+    left_table = _right_in_table;
+    right_table = _left_in_table;
+
+    left_column_id = _right_column_id;
+    right_column_id = _left_column_id;
+  }
+
+  auto left_type_string = left_table->column_type(left_column_id);
+  auto right_type_string = right_table->column_type(right_column_id);
+
+  _pos_list_left = std::make_shared<PosList>();
+  _pos_list_right = std::make_shared<PosList>();
+
+  _is_outer_join = (_mode == JoinMode::Left || _mode == JoinMode::Right || _mode == JoinMode::Outer);
+
+  // Scan all chunks from left input
+  for (ChunkID chunk_id_left = ChunkID{0}; chunk_id_left < left_table->chunk_count(); ++chunk_id_left) {
+    auto column_left = left_table->get_chunk(chunk_id_left).get_column(left_column_id);
+
+    // for Outer joins, remember matches on the left side
+    std::vector<bool> left_matches;
+
+    if (_is_outer_join) {
+      left_matches.resize(column_left->size());
+    }
+
+    // Scan all chunks for right input
+    for (ChunkID chunk_id_right = ChunkID{0}; chunk_id_right < right_table->chunk_count(); ++chunk_id_right) {
+      auto column_right = right_table->get_chunk(chunk_id_right).get_column(right_column_id);
+
+      resolve_data_and_column_type(left_type_string, *column_left, [&](auto left_type, auto& typed_left_column) {
+        resolve_data_and_column_type(right_type_string, *column_right, [&](auto right_type, auto& typed_right_column) {
+          using LeftType = typename decltype(left_type)::type;
+          using RightType = typename decltype(right_type)::type;
+
+          // make sure that we do not compile invalid versions of these lambdas
+          constexpr auto left_is_string_column = (std::is_same<LeftType, std::string>{});
+          constexpr auto right_is_string_column = (std::is_same<RightType, std::string>{});
+
+          constexpr auto neither_is_string_column = !left_is_string_column && !right_is_string_column;
+          constexpr auto both_are_string_columns = left_is_string_column && right_is_string_column;
+
+          // clang-format off
+          if constexpr (neither_is_string_column || both_are_string_columns) {
+            auto iterable_left = create_iterable_from_column<LeftType>(typed_left_column);
+            auto iterable_right = create_iterable_from_column<RightType>(typed_right_column);
+
+            iterable_left.with_iterators([&](auto left_it, auto left_end) {
+              iterable_right.with_iterators([&](auto right_it, auto right_end) {
+                with_operator(_scan_type, [&](auto comparator) {
```

Comment:
Even in the same line you use two different terms :D 
I like "comparator" better, because it is more precise. An "operator" could also be `operator+`.

Reply:
I guess I always thought of the lambda as the "operator", but you are right. `with_comparator` makes more sense